eslint.config.js

```diff
@@ -0,0 +1,82 @@
+const eslint = require('@eslint/js');
+const tseslint = require('typescript-eslint');
+const tsdoc = require('eslint-plugin-tsdoc');
+const jest = require('eslint-plugin-jest');
+const globals = require('globals');
+const prettier = require('eslint-config-prettier');
+
+/** @type {import('eslint').Linter.Config[]} */
+module.exports = [
+  {
+    ignores: ['dist/', 'tests/env/', 'coverage/', 'playgrounds/', 'docs/'],
+  },
+  // Standard linting for js files
+  {
+    files: ['**/*.js'],
+    languageOptions: { sourceType: 'script', globals: globals.node },
+    plugins: { eslint },
+    rules: eslint.configs.recommended.rules,
+  },
+  // TypeScript linting for ts files
+  ...tseslint.configs.recommendedTypeChecked.map((config) => ({
```

Comment:
Now TypeScript rules are also applied to test files, which wasn't the case before.